meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -35,7 +42,101 @@ export const ShowStyleVariantsSettings = withTranslation()(
 
 			this.state = {
 				editedMappings: [],
+				uploadFileKey: Date.now(),
+			}
+		}
+		importVariants(e: React.ChangeEvent<HTMLInputElement>) {
+			const { t } = this.props
+
+			const file = e.target.files ? e.target.files[0] : null
+			if (!file) {
+				return
+			}
+
+			const reader = new FileReader()
+			reader.onload = (e2) => {
+				this.setState({
+					uploadFileKey: Date.now(),
+				})
+
+				const uploadFileContents = (e2.target as any).result
+
+				let newVariants: Array<ShowStyleVariant> = []
+				try {
+					newVariants = JSON.parse(uploadFileContents)
+					if (!_.isArray(newVariants)) {
+						throw new Error('Imported file did not contain an array')
+					}
+				} catch (error) {
+					NotificationCenter.push(
+						new Notification(
+							undefined,
+							NoticeLevel.WARNING,
+							t('Failed to import new showstyle variants: {{errorMessage}}', { errorMessage: error + '' }),
+							'VariantSettings'
+						)
+					)
+					return
+				}
+
+				_.forEach(newVariants, (entry) => {
+					MeteorCall.showstyles
+						.insertShowStyleVariantWithBlueprint(
+							entry.showStyleBaseId,
+							entry.blueprintConfig,
+							entry._id,
+							entry._rundownVersionHash,
+							entry.name
+						)
+						.catch((error) => {
+							NotificationCenter.push(
+								new Notification(
+									undefined,
+									NoticeLevel.WARNING,
+									t('Failed to import Variant ' + entry.name + '. Make sure it is not already imported.', {
+										errorMessage: error + '',
+									}),
+									'VariantSettings'
+								)
+							)
+						})
+				})
 			}
+			reader.readAsText(file)
+		}
+		copyVariant = (showStyleVariant: ShowStyleVariant) => {
```

Comment:
This is very close to what is happening in your forEach in `importVariants`. Consider if you could refactor away duplication.

Reply:
This is outdated with the new implementation, and they use different function to achieve their goal.